File: Problem002/erlang/problem.erl
-module(problem).
-export([main/0]).

main()->
Fib=fibonacci(4000000),
Even=lists:filter(fun(T) -> isOdd(T) end,Fib),
Sum=fun(T,Acc) -> Acc+T end,
lists:foldl(Sum,0,Even).

fibonacci(N)->fibonacci(N,[1|[1]]).
fibonacci(N,List) when hd(List) > N -> List;
fibonacci(N,List)-> 
 [H|T]=List,
  S=hd(T),
  fibonacci(N,[H+S|List]).

isOdd(X) when X rem 2==0 -> true;
isOdd(_) -> false.

File: Problem002/kotlin/problem.kt
fun main(args:Array<String>)
{
    println(Fibonacci(4000000).filter { it % 2==0 }.fold(0,{f,sum->sum+f}))
}
fun Fibonacci(n:Int):List<Int>
{
    var fibonacci=mutableListOf(1,1)
    var i = 2
    while (fibonacci.get(fibonacci.size-1) < n)
    {
        fibonacci.add(fibonacci.get(i-1)+fibonacci.get(i-2))
        i++
    }
    return fibonacci
}